dle/data/management/commands/load_hc_data.py

```diff
@@ -0,0 +1,423 @@
+                logger.error(f"Failed. Retry to request {url}")
+                response = None
+        return response
+
+    def get_and_parse_pdf(self, pdf_url, source_product_number, lp):
+        response = self.requests_with_retries(pdf_url)
+        if not response:
+            logger.error(self.style.ERROR("unable to grab url contents"))
+            self.error_urls[pdf_url] = True
+            return "unable to download pdf"
+
+        filename = default_storage.save(
+            settings.MEDIA_ROOT / "hc.pdf", ContentFile(response.content)
+        )
+        logger.info(f"saved {pdf_url} file to: {filename}")
+        hc_file = settings.MEDIA_ROOT / filename
+        raw_text = self.process_hc_pdf_file(hc_file, source_product_number, lp, pdf_url)
+        # delete the file when done
+        default_storage.delete(filename)
+        return raw_text
+
+    def get_pdf_sections_with_format(self, text, section_format):
+        idx, headers, sections = [], [], []
+        start_idx = 0
+        for section in section_format:
+            for i, line in enumerate(text[start_idx:], start=start_idx):
+                if re.match(section, line.upper()):
+                    start_idx = i + 1  # +1 to start at next line
+                    idx += [i]
+                    headers += [line.strip()]
+                    break
+
+        if len(headers) != 0:
+            idx, headers = filter_headers(idx, headers)
+
+        for n, h in enumerate(headers):
+            if (n + 1) < len(headers):
+                contents = text[idx[n] + 1 : idx[n + 1]]
+            else:
+                contents = text[idx[n] + 1 :]
+            sections += ["\n".join(contents)]
+
+        return headers, sections
+
+    centers = [
+        "Indications",
+        "Indications and Clinical Use",
+        "Contraindications",
+        "Serious Warnings and Precautions Box",
+        "Dosage And Administration",
+        "Overdosage",
+        "Dosage Forms, Strenths, Composition, and Packaging",
+        "Warnings and Precautions",
+        "Warnings",
+        "Precautions",
+        "Adverse Reactions",
+        "Drug Interactions",
+        "Clinical Pharmacology",
+        "Storage, Stability and Disposal",
+        "Special Handling Instructions",
+        "Pharmaceutical Information",
+        "Clinical Trials",
+        "Microbiology",
+        "Non-clinical Toxicology",
+        "Supporting Product Monographs",
+        "Summary Product Information",
+        "Toxicology",
+        "Description",
+    ]
+
+    # improved initial text parsing step so this clustering problem wasn't so messy
+    def get_fixed_header(self, text):
+        # return center with the lowest edit distance,
+        #   or placeholder (last entry) if no there's good match
+        dists = [levdistance(text.lower(), c.lower()) for c in self.centers]
+        # ix = np.argmin(dists)
+        ix = dists.index(min(dists))
+        if dists[ix] > 0.6 * len(text):
```

Comment:
This says, if the edit distance is too great (`>0.6 * len(text)`), return nothing?

Reply:
yup, this is from Caleb's code. Guess it serves as confidence level or some sort